### PR TITLE
Automated cherry pick of #109510: add ILB and NetLB pinhole changes
#111186: Fix firewallNeedsUpdate and firewallRuleEqual for

### PR DESCRIPTION
**Pull request by sugangli**
Cherry pick of #109510 #111186 on release-1.24.

#109510: add ILB and NetLB pinhole changes
#111186: Fix firewallNeedsUpdate and firewallRuleEqual for

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note

```

### CODE REVIEW

**Review by MrHohn — APPROVED** [review]
/ok-to-test

File: staging/src/k8s.io/api/go.mod [hunk 1]
```diff
@@ -7,9 +7,10 @@ go 1.16
 require (
 	github.com/gogo/protobuf v1.3.2
 	github.com/stretchr/testify v1.7.0
-	k8s.io/apimachinery v0.0.0
 )
 
+require k8s.io/apimachinery v0.0.0
```
Comment (cheftako) [inline staging/src/k8s.io/api/go.mod:12]: Why?

**Pull request closed without merging**